libc-test/build.rs

```diff
@@ -3724,6 +3724,8 @@ fn test_linux(target: &str) {
                 => true,
             "SCTP_FUTURE_ASSOC" | "SCTP_CURRENT_ASSOC" | "SCTP_ALL_ASSOC" | "SCTP_PEER_ADDR_THLDS_V2" => true, // linux 5.5+
 
+            "HWTSTAMP_TX_ONESTEP_P2P" if i686 || mips || aarch64_musl => true,
```

Comment:
I think the below should work like others:

```suggestion
            // FIXME: Requires more recent kernel headers
            "HWTSTAMP_TX_ONESTEP_P2P" if mips || musl => true,
```

Reply:
yes, fixed. A couple more linux flavors to get through

Reply:
The condition doesn't seem to be changed. It should `mips || musl` as we use older kernels for these targets.

Reply:
though maybe musl is more relevant than aarch here?

Reply:
We use the same kernel version for _all_ the musl targets across CI: https://github.com/rust-lang/libc/blob/c64bfc5fb36414e02ad7977b32df765876632383/ci/install-musl.sh#L82
So, CPU arch doesn't involve here, we just need to ignore `musl` on the if cond (the same applies to sparc64).

Reply:
so `i686 || mips || musl`, or only `musl` even?

Reply:
Ah, I mean the condition should be `sparc64 || musl`.

Reply:
For sparc64, see:
https://github.com/rust-lang/libc/blob/c64bfc5fb36414e02ad7977b32df765876632383/ci/docker/sparc64-unknown-linux-gnu/Dockerfile#L1-L2

Reply:
i686 itself isn't related to the failure, it fails due to the musl toolchain.

Reply:
makes sense, I've updated it